python/spacetime/store.py

```diff
@@ -162,9 +174,20 @@ def __init__(self, ):
         self.__typename_to_primarykey = {}
         for tp in DATAMODEL_TYPES:
             if tp.__PCC_BASE_TYPE__ or tp.__name__ == "_Join":
-                self.__typename_to_primarykey[tp.Class().__name__] = tp.__primarykey__._name
+                self.__typename_to_primarykey[tp.__realname__] = tp.__primarykey__._name
             else:
-                self.__typename_to_primarykey[tp.Class().__name__] = tp.__ENTANGLED_TYPES__[0].__primarykey__._name
+                self.__typename_to_primarykey[tp.__realname__] = tp.__ENTANGLED_TYPES__[0].__primarykey__._name
+
+    def reload_dms(self):
+        from datamodel.all import DATAMODEL_TYPES
+        self.__base_store.reload_dms(DATAMODEL_TYPES)
+
+        self.__typename_to_primarykey = {}
+        for tp in DATAMODEL_TYPES:
+            if tp.__PCC_BASE_TYPE__ or tp.__name__ == "_Join":
+                self.__typename_to_primarykey[tp.__realname__] = tp.__primarykey__._name
+            else:
+                self.__typename_to_primarykey[tp.__realname__] = tp.__ENTANGLED_TYPES__[0].__primarykey__._name
```

Comment:
Do you want to change it? Or should we commit it like this and you will handle it later?